test/helpers/assertThrow.js

```diff
@@ -4,7 +4,8 @@ function assertError(error, s, message) {
 
 async function assertThrows(codeBlock, message, errorCode) {
     try {
-        await codeBlock()
+        const result = await codeBlock()
+        return assert.equal(result.receipt.status, 0, 'should have failed status')
```

Comment:
Hmm, not sure I like this; the `assert.fail('should have thrown before')` at the end should fail any test that doesn't throw an error.

Not having this bit also lets us not worry about what's being returned from the `codeBlock()`.